src/measure_theory/interval_integral.lean

```diff
@@ -1345,4 +1345,128 @@ theorem integral_deriv_eq_sub (hderiv : ∀ x ∈ interval a b, differentiable_a
 integral_eq_sub_of_has_deriv_at (λ x hx, (hderiv x hx).has_deriv_at) hcont'
   (measurable_deriv f).ae_measurable
 
+/-!
+### Integration by parts
+ -/
+
+lemma self_mem_ae_restrict {s : set ℝ} (hs : is_measurable s):
+  s ∈ (measure.restrict measure_space.volume s).ae :=
+begin
+  rw ae_restrict_eq hs,
+  simp only [exists_prop, filter.mem_principal_sets, filter.mem_inf_sets],
+  exact ⟨univ, filter.univ_mem_sets, s, by simp⟩,
+end
+
+lemma integral_congr_Ioc {f g : ℝ → ℝ} (h : ∀ x ∈ Ioc a b, f x = g x) :
+  ∫ x in Ioc a b, f x = ∫ x in Ioc a b, g x :=
+integral_congr_ae (filter.eventually_of_mem (self_mem_ae_restrict is_measurable_Ioc) h)
+
+/--
+If f and g coincide in the relevant integral, their integrals also coincide.
+-/
+lemma interval_integral_congr {f g : ℝ → ℝ} (h : ∀ x ∈ interval a b, f x = g x) :
+ ∫ x in a..b, f x = ∫ x in a..b, g x :=
+begin
+  unfold interval_integral,
+  unfold interval at h,
+  cases le_total a b with hab hab,
+  {
+    simp only [hab, Ioc_eq_empty, measure.restrict_empty,
+      min_eq_left, max_eq_right, integral_zero_measure, sub_zero] at h ⊢,
+    apply integral_congr_Ioc (λ x hx, h x (mem_Icc_of_Ioc hx)),
+  },
```

Comment:
```suggestion
  { simp only [hab, Ioc_eq_empty, measure.restrict_empty,
      min_eq_left, max_eq_right, integral_zero_measure, sub_zero] at h ⊢,
    apply integral_congr_Ioc (λ x hx, h x (mem_Icc_of_Ioc hx)), },
```

here, and everywhere else. The mathlib style is to not have braces on their own lines.

https://leanprover-community.github.io/contribute/style.html#tactic-mode